In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

## Importing First Dataset

In [2]:
# using dataset from https://www.kaggle.com/devvret/brazil-real-estate-listings

data = pd.read_csv('./properati_br_2016_11_01_properties_rent.csv')

In [3]:
data

created_on operation property_type       place_name  \
0      2013-04-25      rent     apartment     Praia Grande   
1      2013-04-25      rent     apartment       Alphaville   
2      2013-04-25      rent         house           Centro   
3      2013-04-25      rent     apartment       Alphaville   
4      2013-04-25      rent     apartment          Barueri   
...           ...       ...           ...              ...   
97348  2016-11-01      rent     apartment      Vila Adyana   
97349  2016-11-01      rent     apartment  Jardim Aquárius   
97350  2016-11-01      rent     apartment  Jardim Aquárius   
97351  2016-11-01      rent     apartment       Votorantim   
97352  2016-11-01      rent     apartment       Copacabana   

                                 place_with_parent_names  geonames_id  \
0                        |Brasil|São Paulo|Praia Grande|          NaN   
1                  |Brasil|São Paulo|Barueri|Alphaville|          NaN   
2                      |Brasil|São Paulo|Jundiaí|Centro|          NaN   
3                  |Brasil|São Paulo|Barueri|Alphaville|          NaN   
4                             |Brasil|São Paulo|Barueri|          NaN   
...                                                  ...          ...   
97348  |Brasil|São Paulo|São José dos Campos|Vila Ady...          NaN   
97349  |Brasil|São Paulo|São José dos Campos|Jardim A...          NaN   
97350  |Brasil|São Paulo|São José dos Campos|Jardim A...          NaN   
97351                      |Brasil|São Paulo|Votorantim|          NaN   
97352  |Brasil|Rio de Janeiro|Rio de Janeiro|Copacabana|          NaN   

                            lat_lon        lat        lon   price  ...  \
0                               NaN        NaN        NaN   900.0  ...   
1                               NaN        NaN        NaN  3200.0  ...   
2                               NaN        NaN        NaN  1800.0  ...   
3                               NaN        NaN        NaN     NaN  ...   
4                               NaN        NaN        NaN  1800.0  ...   
...                             ...        ...        ...     ...  ...   
97348                           NaN        NaN        NaN  1600.0  ...   
97349                           NaN        NaN        NaN  2000.0  ...   
97350                           NaN        NaN        NaN  1900.0  ...   
97351                           NaN        NaN        NaN   900.0  ...   
97352  POINT(-43.188034 -22.966091) -22.966091 -43.188034  1800.0  ...   

      price_usd_per_m2  price_per_m2  floor  rooms  expenses  \
0                  NaN           NaN    NaN    1.0       NaN   
1                  NaN           NaN    NaN    3.0       NaN   
2                  NaN           NaN    NaN    2.0       NaN   
3                  NaN           NaN    NaN    NaN       NaN   
4                  NaN           NaN    NaN    NaN       NaN   
...                ...           ...    ...    ...       ...   
97348              NaN           NaN    NaN    NaN       NaN   
97349              NaN           NaN    NaN    NaN       NaN   
97350              NaN           NaN    NaN    NaN       NaN   
97351              NaN           NaN    NaN    1.0     301.0   
97352        17.020909     54.545455    NaN    1.0     650.0   

                                           properati_url  \
0      http://praia-grande-sao-paulo.properati.com.br...   
1      http://alphaville-barueri.properati.com.br/q8z...   
2      http://centro-jundiai.properati.com.br/qe2_alu...   
3      http://alphaville-barueri.properati.com.br/s9p...   
4      http://barueri.properati.com.br/s9u_aluga-se_a...   
...                                                  ...   
97348  http://vila-adyana.properati.com.br/2hk0a_alug...   
97349  http://jardim-aquarius.properati.com.br/2hk0b_...   
97350  http://jardim-aquarius.properati.com.br/2hk0c_...   
97351  http://votorantim-sao-paulo.properati.com.br/2...   
97352  http://copacabana.properati.com.br/2hk0j_aluga...   

      

In [4]:
data.describe()

geonames_id           lat           lon          price  \
count          3.0  46030.000000  46030.000000   92334.000000   
mean     4164138.0    -22.861939    -46.707036    3935.703554   
std            0.0      3.585125      2.672557    9325.828330   
min      4164138.0    -33.692432    -97.497955       0.000000   
25%      4164138.0    -23.619106    -47.024731    1140.000000   
50%      4164138.0    -23.562963    -46.671908    1900.000000   
75%      4164138.0    -23.030475    -46.567808    3600.000000   
max      4164138.0     25.860630    -34.813914  470000.000000   

       price_aprox_local_currency  price_aprox_usd  surface_total_in_m2  \
count                92334.000000     92334.000000         15848.000000   
mean                  3928.693581      1228.255667           234.658695   
std                   9197.214373      2875.387441          2444.656264   
min                      0.000000         0.000000             0.000000   
25%                   1127.850000       352.610000             0.000000   
50%                   1918.200000       599.700000            79.000000   
75%                   3593.240000      1123.380000           180.000000   
max                 353358.360000    110472.820000        200000.000000   

       surface_covered_in_m2  price_usd_per_m2  price_per_m2          floor  \
count           75584.000000      70415.000000  70415.000000    5285.000000   
mean              152.538500         16.288983     52.128047      49.528855   
std               645.357711        212.996846    681.577383    3043.289924   
min                 0.000000          0.028084      0.090000       1.000000   
25%                52.000000          5.200800     16.666667       3.000000   
50%                80.000000          7.801250     25.000000       6.000000   
75%               150.000000         12.482000     40.000000      10.000000   
max             75000.000000      28708.730000  92000.000000  221240.000000   

              rooms      expenses  
count  58763.000000  3.177300e+04  
mean       2.158161  9.759202e+02  
std        2.192546  9.845683e+03  
min        1.000000  1.000000e+00  
25%        1.000000  3.500000e+02  
50%        1.000000  5.560000e+02  
75%        2.000000  9.530000e+02  
max       40.000000  1.490000e+06

In [5]:
data.isnull().sum()

created_on                        0
operation                         0
property_type                     0
place_name                        0
place_with_parent_names           0
geonames_id                   97350
lat_lon                       51323
lat                           51323
lon                           51323
price                          5019
currency                       5020
price_aprox_local_currency     5019
price_aprox_usd                5019
surface_total_in_m2           81505
surface_covered_in_m2         21769
price_usd_per_m2              26938
price_per_m2                  26938
floor                         92068
rooms                         38590
expenses                      65580
properati_url                     0
description                       0
title                             0
image_thumbnail                5134
location                      51323
dtype: int64

Looks like the latitude and longitude are mostly empty. We might have to not use that data and focus on neighborhoods if we use this dataset.

## São Paulo

In [6]:
data['place_with_parent_names'].value_counts()

|Brasil|São Paulo|Santo André|                       2322
|Brasil|São Paulo|São Paulo|Moema|                   2214
|Brasil|São Paulo|Praia Grande|                      1784
|Brasil|São Paulo|São Paulo|                         1763
|Brasil|São Paulo|São Paulo|Jardim Paulista|         1628
                                                     ... 
|Brasil|São Paulo|Bauru|Jardim Dona Sarah|              1
|Brasil|Paraná|Iguatu|                                  1
|Brasil|São Paulo|Marília|Barbosa|                      1
|Brasil|Paraíba|Santo André|Jardim Alzira Franco|       1
|Brasil|São Paulo|Ribeirão Preto|Vila Amélia|           1
Name: place_with_parent_names, Length: 3728, dtype: int64

In [7]:
# code found at 
# https://stackoverflow.com/questions/44237170/how-to-search-a-partial-string-in-the-whole-dataframe-using-pandas

mask = data.applymap(lambda x: 'São Paulo|São Paulo' in str(x))

In [8]:
data1 = data[mask.any(axis=1)]
data1.shape

(39191, 25)

Of the 97,353 listings in the dataset, 39,191 come from São Paulo, the city (it's also the name of a state, so it's important to specify).

In [9]:
data1['place_name'].value_counts()

Moema                  2214
São Paulo              1763
Jardim Paulista        1628
Vila Nova Conceição    1289
Itaim Bibi             1248
                       ... 
Jardim Everest            1
Vila Fernandes            1
Vila Tiradentes           1
Jardim Itápolis           1
Vila Santos               1
Name: place_name, Length: 538, dtype: int64

Looks like we have 538 different places in São Paulo.

According to [Wikipedia](#https://en.wikipedia.org/wiki/Subdivisions_of_S%C3%A3o_Paulo), there are 32 boroughs in the city, and a total of 96 districts between them. There may be a number of neighborhoods within each district. This may make mapping difficult.

In [10]:
data1['property_type'].value_counts()

apartment    26303
store         6649
house         6077
PH             162
Name: property_type, dtype: int64

Most of the listings in Sao Paulo are apartments, with 26,303 listings. There's an almost equal number of stores and houses, with 6649 stores and 6077 houses.

In [20]:
data1.isnull().sum()

created_on                        0
operation                         0
property_type                     0
place_name                        0
place_with_parent_names           0
geonames_id                   39191
lat_lon                       17468
lat                           17468
lon                           17468
price                          2878
currency                       2879
price_aprox_local_currency     2878
price_aprox_usd                2878
surface_total_in_m2           31658
surface_covered_in_m2          5534
price_usd_per_m2               8804
price_per_m2                   8804
floor                         36944
rooms                         14108
expenses                      23342
properati_url                     0
description                       0
title                             0
image_thumbnail                2782
location                      17468
dtype: int64

## Rio de Janiero

In [11]:
mask_rio = mask = data.applymap(lambda x: 'Rio de Janeiro' in str(x))

In [12]:
data2 = data[mask_rio.any(axis=1)]
data2.shape

(3603, 25)

Only about 3,603 listings are from Rio de Janiero, which might not be enough data to work with.

## Importing Second Dataset

In [13]:
# using data for Sāo Paulo from https://www.kaggle.com/argonalyst/sao-paulo-real-estate-sale-rent-april-2019

df = pd.read_csv('./sao-paulo-properties-april-2019.csv')

In [14]:
df.shape

(13640, 16)

It's only 13,640 listings long, which isn't too many, comparitvely.

In [15]:
df.dtypes

Price                 int64
Condo                 int64
Size                  int64
Rooms                 int64
Toilets               int64
Suites                int64
Parking               int64
Elevator              int64
Furnished             int64
Swimming Pool         int64
New                   int64
District             object
Negotiation Type     object
Property Type        object
Latitude            float64
Longitude           float64
dtype: object

In [16]:
df.isnull().sum()

Price               0
Condo               0
Size                0
Rooms               0
Toilets             0
Suites              0
Parking             0
Elevator            0
Furnished           0
Swimming Pool       0
New                 0
District            0
Negotiation Type    0
Property Type       0
Latitude            0
Longitude           0
dtype: int64

Oh wow, there's no nulls in this data, we can actually use it for longitude and latitude.

In [17]:
df['Property Type'].value_counts()

apartment    13640
Name: Property Type, dtype: int64

Okay, so it's all apartments.

In [18]:
# removing Sāo Paulo from the end of all the District values

district_split = df["District"].str.split("/", n = 1, expand = True) 
df['District'] = district_split[0]
df['District'].value_counts()

Moema              293
Mooca              288
Brás               255
Bela Vista         250
Brooklin           250
                  ... 
Jardim São Luis     44
São Rafael          44
Grajaú              37
São Domingos        35
Perus               30
Name: District, Length: 96, dtype: int64

Looks like this dataset has listings from all 36 wards. It doesn't go into detail about the specific neighborhoods, but paired with geolocation data, it may be enough to use.

In [19]:
df['New'].value_counts()

0    13427
1      213
Name: New, dtype: int64

This is interesting, but may not be useful. Only 213 of the listings in this dataset are new.

The one thing this dataset doesn't include is the date of the listing. I assume these were all available in April 2019, like the Kaggle page suggests. The project guidelines suggested getting years worth of data. Just one month likely isn't enough of a timeframe to be able to correctly find informal settlements. There could just be a lack of listings in particular areas that month.